cpp/src/arrow/adapters/orc/adapter.h

```diff
@@ -56,55 +66,114 @@ class ARROW_EXPORT ORCFileReader {
   /// \brief Return the schema read from the ORC file
   ///
   /// \param[out] out the returned Schema object
+  ARROW_DEPRECATED("Deprecated in 6.0.0.")
```

Comment:
This is good, but it would be better to make the message a bit more informative. How about:
```c++
ARROW_DEPRECATED("Deprecated in 6.0.0. Use Result-returning overload instead.")
```